ledger/src/blockstore_processor.rs

```diff
@@ -921,21 +923,41 @@ pub fn confirm_slot(
 
     let verify_transaction = {
         let bank = bank.clone();
-        move |versioned_tx: VersionedTransaction| -> Result<SanitizedTransaction> {
-            bank.verify_transaction(versioned_tx, skip_verification)
+        move |versioned_tx: VersionedTransaction,
+              verification_mode: TransactionVerificationMode|
+              -> Result<SanitizedTransaction> {
+            bank.verify_transaction(versioned_tx, verification_mode)
         }
     };
-    let check_start = Instant::now();
-    let mut entries = entry::verify_transactions(entries, Arc::new(verify_transaction))?;
-    let transaction_duration_us = timing::duration_as_us(&check_start.elapsed());
+
+    let mut check_result = entry::start_verify_transactions(
+        entries,
+        skip_verification,
+        recyclers.clone(),
+        Arc::new(verify_transaction),
+    );
+
+    // If the CPU Entry signature verification path is used
+    // (for example, when no GPU acceleration API is in use)
+    // we will have our verification results immediately,
+    // so check if the verification has failed, and if so
+    // return early. Otherwise, the result will still be pending
+    // as the signature verification is running on the GPU
+    if check_result.status() == EntryVerificationStatus::Failure {
+        warn!("Ledger proof of history failed at slot: {}", slot);
+        return Err(BlockError::InvalidEntryHash.into());
```

Comment:
We used to bubble up `TransactionError` which gets converted to a `BlockstoreProcessorError::InvalidTransaction` error. Can you preserve that behavior? `InvalidEntryHash` isn't relevant here.

Reply:
Done